itest/litd_node.go

```diff
@@ -0,0 +1,1704 @@
+package itest
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/chanbackup"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/watchtowerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/wtclientrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"gopkg.in/macaroon.v2"
+)
+
+const (
+	// logPubKeyBytes is the number of bytes of the node's PubKey that will
+	// be appended to the log file name. The whole PubKey is too long and
+	// not really necessary to quickly identify what node produced which
+	// log file.
+	logPubKeyBytes = 4
+
+	// trickleDelay is the amount of time in milliseconds between each
+	// release of announcements by AuthenticatedGossiper to the network.
+	trickleDelay = 50
+
+	// listenerFormat is the format string that is used to generate local
+	// listener addresses.
+	listenerFormat = "127.0.0.1:%d"
+)
+
+var (
+	// numActiveNodes is the number of active nodes within the test network.
+	numActiveNodes    = 0
+	numActiveNodesMtx sync.Mutex
+)
+
+// generateListeningPorts returns four ints representing ports to listen on
+// designated for the current lightning network test. This returns the next
+// available ports for the p2p, rpc, rest and profiling services.
+func generateListeningPorts(cfg *NodeConfig) {
+	if cfg.P2PPort == 0 {
+		cfg.P2PPort = lntest.NextAvailablePort()
+	}
+	if cfg.RPCPort == 0 {
+		cfg.RPCPort = lntest.NextAvailablePort()
+	}
+	if cfg.RESTPort == 0 {
+		cfg.RESTPort = lntest.NextAvailablePort()
+	}
+	if cfg.ProfilePort == 0 {
+		cfg.ProfilePort = lntest.NextAvailablePort()
+	}
+}
+
+type NodeConfig struct {
```

Comment:
Is there anything we need custom here, or can we just import the current `lnd` instance for something like this?